urlc4364d0d12_ahmedelmorsy_cafetria_simulation_tgz-pJ8-main_MainJ8/src/main/Simulation.java

```diff
@@ -0,0 +1,363 @@
+package main;
+
+import java.io.File;
+import java.util.Random;
+
+import simulation.global.Const;
+import simulation.global.Event;
+import simulation.global.EventsQueue;
+import simulation.global.SimulationClk;
+import simulation.global.Statistics;
+import simulation.logging.FileLogger;
+import simulation.queue.CustomServer;
+import simulation.queue.Customer;
+import simulation.queue.InfServersQueueSystem;
+import simulation.queue.QueueSystem;
+import simulation.queue.UniformServer;
+import simulation.random.ExponentialRandom;
+
+public class Simulation {
+
+    private static final int MAX_TIME = 6300;
+    private static final int INTERARRIVAL_AVG = 30;
+    private static Random groupsRand = new Random();
+    private static Random routeRand = new Random();
+
+    private QueueSystem hotfoodSys;
+    private QueueSystem sandwichesSys;
+    private QueueSystem drinkSys;
+    private QueueSystem[] cashiers;
+
+    private ExponentialRandom arrivalRand;
+
+    public Simulation(int hotfoodEmp, int sandwichesEmp, int cashiersNum,
+            String traceFile) {
+        SimulationClk.clock = 0;
+
+        this.arrivalRand = new ExponentialRandom(INTERARRIVAL_AVG);
+        init(hotfoodEmp, sandwichesEmp, cashiersNum);
+        Statistics.trace = new FileLogger(new File(traceFile));
+    }
+
+    private void init(int hotfoodEmp, int sandwichesEmp, int cashiersNum) {
+        UniformServer hotfoodServ = new UniformServer(Const.HOTFOOD_SERVER, 50,
+                120, 20, 40, hotfoodEmp);
+        UniformServer sandwichesServ = new UniformServer(Const.SANDWICH_SERVER,
+                60, 180, 5, 15, sandwichesEmp);
+        UniformServer drinksServ = new UniformServer(Const.DRINKS_SERVER, 5,
+                20, 5, 10, 1);
+
+        this.hotfoodSys = new QueueSystem(Const.HOTFOOD_SERVER, hotfoodServ, -1);
+        this.sandwichesSys = new QueueSystem(Const.SANDWICH_SERVER,
+                sandwichesServ, -1);
+        this.drinkSys = new InfServersQueueSystem(Const.DRINKS_SERVER,
+                drinksServ, -1);
+
+        this.cashiers = new QueueSystem[cashiersNum];
+        for (int i = 0; i < cashiersNum; i++) {
+            CustomServer temp = new CustomServer(Const.CASHIER_SERVER);
+            cashiers[i] = new QueueSystem(Const.CASHIER_SERVER, temp, -1);
+        }
+    }
+
+    public void run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 113 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 66-161 into a separate method.